shm/cargo.cpp

```diff
@@ -0,0 +1,15 @@
+#include "cargo.hpp"
+
+Cargo();
+Cargo(uint8_t amount, std::string name, uint8_t base_price)
+    : amount_(amount), name_(name), base_price_(base_price)
+{}
+~Cargo() {;} 
+
+std::string getName() const { return name_; }
+uint8_t getAmount() const { return amount_; }
+uint8_t getBase_price() const { return base_price_; }
```

Comment:
every where with methods should be :

std::string Cargo::getName() const { return name_; }
uint8_t Cargo::getAmount() const { return amount_; }
uint8_t Cargo::getBase_price() const { return base_price_; }

in my opinion